Android.mk

```diff
@@ -265,11 +265,20 @@ HYBRIS_UPDATER_UNPACK := $(LOCAL_BUILD_MODULE)
 
 .PHONY: hybris-hal hybris-common
 
-hybris-common: bootimage hybris-updater-unpack hybris-updater-script hybris-recovery hybris-boot servicemanager logcat updater init adb adbd linker libc libEGL libGLESv1_CM libGLESv2
+HYBRIS_COMMON_TARGETS := bootimage hybris-updater-unpack hybris-updater-script hybris-recovery hybris-boot servicemanager logcat updater init adb adbd linker libc libEGL libGLESv1_CM libGLESv2 libsurfaceflinger libsf_compat_layer
```

Comment:
This means libsurfaceflinger and its compat layer are now going to build for all older adaptations. It's all good, as long as their build products won't end up under /usr/libexec/droid-hal :)

Reply:
my reasoning behind this was, in some hybris-14.1 adaptations some people are using surfaceflinger to render, so i included it for all. I also included a systemd service and *.rc workaround that would start surfaceflinger if **and only if** it is required in the compositor configuration. that way community adaptations could have surfaceflinger rendering enabled/disabled by changing an env variable (since the patched libs will be included by this) and thus simplifying the porting effort a bit, though we still would need to make the systemd service and *.rc fix a part of the upstream dhd submodule and not the device specific one.

if it's better to just do it for Android 8 and above then i will change it.

Though this introduces an issue: libhybris needs to exist in the $ANDROID_ROOT before build_packages.sh has been run. If we don't handle this case specifically in build_packages.sh + manifest.xml we will have an issue with either obs, or local builds.

Reply:
We'll end up handling libhybris case in build_packages.sh for now.